lib/measured/rails/active_record.rb

```diff
@@ -1,9 +1,71 @@
-module Measured::Rails
+module Measured::Rails::ActiveRecord
   extend ActiveSupport::Concern
 
   module ClassMethods
-    def measured
+    def measured(measured_class, *fields)
+      options = fields.extract_options!
+      options = {}.merge(options)
 
+      measured_class = measured_class.constantize if measured_class.is_a?(String)
+
+      raise Measured::Rails::Error, "Class must be an instance of Measured::Measurable" unless measured_class.ancestors.include?(Measured::Measurable)
+
+      options[:class] = measured_class
+
+      fields.map(&:to_sym).each do |field|
+        raise Measured::Rails::Error, "The field #{ field } has already been measured" if measured_fields.keys.include?(field)
+        measured_fields[field] = options
+
+        # Reader to retrieve measured object
+        define_method(field) do
+          value = public_send("#{ field }_value")
+          unit = public_send("#{ field }_unit")
+
+          return nil unless value && unit
+
+          instance = instance_variable_get("@measured_#{ field }")
+          new_instance = begin
+            measured_class.new(value, unit)
+          rescue Measured::UnitError
+            nil
+          end
+
+          if instance && instance == new_instance
+            instance
+          else
+            instance_variable_set("@measured_#{ field }", new_instance)
+          end
+        end
+
+        # Writer to assign measured object
+        define_method("#{ field }=") do |incoming|
```

Comment:
So just to sanity-check my understanding of this, you'd need to do assignment through a Measured object with units, with something like e.g.:

variant.height = Measured::Length.new(20, :in)

but if you call it with anything else then unit and value are set to nil, which presumably will fail validation, if you've asked for it?